src/tracks/MidiClip.cpp

```diff
@@ -355,16 +365,20 @@ void MidiClip::setType( MidiClipTypes _new_clip_type )
 void MidiClip::checkType()
 {
 	NoteVector::Iterator it = m_notes.begin();
-	while( it != m_notes.end() )
+
+	// If all notes are StepNotes, we have a BeatClip
+	bool beatClip = true;
+	while (it != m_notes.end())
```

Comment:
Since ``QVector`` provides ``begin()`` and ``end()`` members, we could use a ranged-based for loop (Or even better, use ``std::find_if`` or ``std::count_if`` to make this code far more compact. Any one works, although ``std::find_if`` expresses intent more clearly IMO).

Reply:
From what I read, since the conditional is a "different-than" I would have to use `std::not1` with a conditional bool method, which I think kills a bit the purpose of simplifying this code. But I changed it to use a range-based loop!

Reply:
> From what I read, since the conditional is a "different-than" I would have to use `std::not1` with a conditional bool method, which I think kills a bit the purpose of simplifying this code. But I changed it to use a range-based loop!

It should be as simple as writing (not tested):
```cpp
auto beatClip = std::find_if(m_notes.begin(), m_notes.end(), [](auto note) { return note->type() != Note::StepNote }) == m_notes.end();
```

So, if a ``Note`` is found that is not a ``StepNote``, then the iterator returned would not be equal to ``m_notes.end()`` and thus the whole expression would return false. Otherwise, if every note is a ``StepNote``, then the iterator returned is ``m_notes.end()``, and the expression would return true.